src/adaptive_graph_of_thoughts/domain/services/neo4j_utils.py

```
@@ -254,51 +259,38 @@ async def delete_node(node_id: str) -> list[Record]:
-        if not key.replace('_', '').replace('-', '').isalnum():
-            raise ValueError(
-                f"Invalid property name: {key}. "
-                "Property names must be alphanumeric with underscores/hyphens only."
-            )
-
     # Validate node IDs
     try:
         from_id_int = int(from_id)
         to_id_int = int(to_id)
     except ValueError:
         raise ValueError("Invalid node IDs. Both from_id and to_id must be valid integers.")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)